### PR TITLE
Feature/177 create library

### PR DESCRIPTION
**Pull request by 0x64746b**
Hey guys,

I had a stab at #177. It's been a long time since I touched anything in Android or even Java. So any feedback from style conventions to how I integrated with the existing features to where I put stuff is highly appreciated.

As a rule of thumb I tried to be consistent wrt the existing style and create a minimalist diff.

If we can hone this to a point where it can be merged, I'd be interested in building upon it to add the ability to further manage (delete, rename, ...) libraries.

Thanks in advance
dtk

### CODE REVIEW

File: app/src/main/res/layout/dialog_new_lib.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        android:inputType="text" />
+
+    <EditText
+        android:id="@+id/new_lib_password"
+        android:hint="@string/passwd_hint"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="@drawable/edit_text_holo_light"
+        android:textSize="@dimen/dialog_msg_txt_size"
+        android:inputType="text" />
```
Comment (Logan676) [inline app/src/main/res/layout/dialog_new_lib.xml]: should use `android:inputType="textPassword"` instead

File: app/src/main/java/com/seafile/seadroid2/SeafConnection.java [hunk 2]
```diff
@@ -1032,6 +1032,21 @@ private String uploadBlocksCommon(String link, String repoID, String dir,
         }
     }
 
+    public void createNewLib(String libName, String description, String password) throws SeafException {
```
Comment (Logan676) [inline app/src/main/java/com/seafile/seadroid2/SeafConnection.java]: we use `repo` internally to describe library, so better to refactor those names.

File: app/src/main/java/com/seafile/seadroid2/ui/dialog/NewLibDialog.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+import android.view.View;
+import android.view.WindowManager;
+import android.widget.EditText;
+
+import com.seafile.seadroid2.R;
+import com.seafile.seadroid2.SeafException;
+import com.seafile.seadroid2.account.Account;
+import com.seafile.seadroid2.data.DataManager;
+
+class NewLibTask extends TaskDialog.Task {
```
Comment (Logan676) [inline app/src/main/java/com/seafile/seadroid2/ui/dialog/NewLibDialog.java]: better to rename it as `CreateRepoTask`

File: app/src/main/java/com/seafile/seadroid2/ui/dialog/NewLibDialog.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    protected void runTask() {
+        try {
+            mDataManager.createNewLib(mLibName, mDescription, mPassword);
+        } catch (SeafException e) {
+            setTaskException(e);
+        }
+    }
+}
+
+public class NewLibDialog extends TaskDialog {
```
Comment (Logan676) [inline app/src/main/java/com/seafile/seadroid2/ui/dialog/NewLibDialog.java]: CreateRepoDialog

File: app/src/main/res/layout/dialog_new_lib.xml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        android:id="@+id/new_lib_name"
+        android:hint="@string/name_hint"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="@drawable/edit_text_holo_light"
+        android:textSize="@dimen/dialog_msg_txt_size"
+        android:inputType="text" />
+
+    <EditText
+        android:id="@+id/new_lib_description"
```
Comment (Logan676) [inline app/src/main/res/layout/dialog_new_lib.xml]: better to remove the element, like seahub, we don\`t ask user to input descriptions any more.

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -131,13 +134,15 @@
         <item>Download folder</item>
         <item>Download folder and subfolders</item>
     </string-array>
+    <string name="add_lib">Add</string>
```
Comment (Logan676) [inline app/src/main/res/values/strings.xml]: better to use `New Library`

**Comment by Logan676** [comment]
awesome patch by the way 👍

**Comment by 0x64746b** [comment]
@Logan676 Thx for the feedback!

> should use android:inputType="textPassword" instead

I [considered that](https://github.com/haiwen/seadroid/pull/557/commits/47dec4262f99ea59db1aa9eee7e6e817e536fb13#diff-2d91b11af6c46b19c32514a88a50e53fR46). Do you want me to add a second password field to check for consistency? Or set the input type to `textVisiblePassword`?

**Comment by Logan676** [comment]
@0x64746b 
The prototype when create new libraries like below, 
create_encrypted_lib
create_normal_lib

**Comment by 0x64746b** [comment]
@Logan676 Yeah, that basically matches the web interface.

Where are those screenshots from? The iOS client? Or is it just a mockup?

**Comment by loganguo** [comment]
It is just a mockup.
In fact the create-new-lib feature doesn\`t finished yet until someone has implemented the ui stuff.

**Comment by Logan676** [comment]
It is just a mockup.
In fact the create-new-lib feature doesn`t finished yet until someone has implemented the ui stuff.

**Pull request merged**